include/pops/cauchy_kernel.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * PoPS model - cauchy dispersal kernel
+ *
+ * Copyright (C) 2015-2020 by the authors.
+ *
+ * Authors: Margaret Lawrimore malawrim ncsu edu
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#ifndef POPS_CAUCHY_KERNEL_HPP
+#define POPS_CAUCHY_KERNEL_HPP
+
+#include "kernel_types.hpp"
+#include "raster.hpp"
+
+#include <random>
+
+namespace pops {
+
+using std::pow;
+using std::tan;
+
+/*! Dispersal kernel for cauchy distribution
+ *  class utilized by RadialKernel and DeterministicKernel
+ */
+class CauchyKernel
+{
+protected:
+    double s;
+    std::cauchy_distribution<double> cauchy_distribution;
+
+public:
+    CauchyKernel(double ss) : s(ss), cauchy_distribution(0, s) {}
+
+    /*!
+     *  Returns random value from cauchy distribution
+     *  Used by RadialKernel to determine location of spread
+     *  @param generator uniform random number generator
+     *  @return value from cauchy distribution
+     */
+    template<class Generator>
+    double random(Generator& generator)
+    {
+        return std::abs(cauchy_distribution(generator));
+    }
```

Comment:
I wanted to explicitly name it random so that it would be more clear that it is used only by radial kernel while the other two functions are used by deterministic kernel. But, I can also name it something else or change it to operator() if that would be more clear.